packages/wallet-connectors/src/WalletConnection.ts

```diff
@@ -70,12 +70,16 @@ export function typeSchema(schema: Buffer): TypeSchema {
 }
 
 function schemaAsBuffer(schemaBase64: string) {
-    const res = toBuffer(schemaBase64, 'base64');
-    // Check round-trip. This requires the provided schema to be properly padded.
-    if (res.toString('base64') !== schemaBase64) {
-        throw new Error(`provided schema '${schemaBase64}' is not valid base64`);
+    const res = Buffer.from(schemaBase64, 'base64');
+    // Check round-trip. We allow for unpadded base64.
+    const serialized = res.toString('base64');
+    if (serialized.startsWith(schemaBase64)) {
+        const padding = serialized.slice(schemaBase64.length);
+        if (['==', '=', ''].includes(padding)) {
+            return res;
+        }
     }
-    return res;
+    throw new Error(`The provided schema '${schemaBase64}' is not valid base64`);
```

Comment:
This is actually not entirely correct (neither was it before) because base64 isn't fully deterministic. For example `aa==` and `aQ==` mean the same thing.

It's also much more efficient to compare the encoded/decoded lengths instead of round-tripping. As I understand [this](https://stackoverflow.com/questions/13378815/base64-length-calculation),`n` bytes should encode to `4*ceil(n/3)` characters with padding, `ceil(4*n/3)` without it.

So I believe we can replace the function with the following:
```typescript
export function schemaAsBuffer(schemaBase64: string) {
    let unpaddedLen = schemaBase64.length;
    if (schemaBase64.charAt(unpaddedLen - 1) === '=') unpaddedLen--;
    if (schemaBase64.charAt(unpaddedLen - 1) === '=') unpaddedLen--;
    const res = toBuffer(schemaBase64, 'base64');
    if (unpaddedLen !== Math.ceil(4 * res.length / 3)) {
        throw new Error(`provided schema '${schemaBase64}' is not valid base64`);
    }
    return res;
}
```

The reason I'm comparing the unpadded length is to avoid accepting invalid characters as padding.

Reply:
Won't this panic on an empty string?

Reply:
It would in sane languages but in JS `str.chatAt` returns the empty string when out of bounds.

Reply:
I have applied the suggestion.